src/pane.coffee

```diff
@@ -514,7 +514,8 @@ class Pane extends Model
 
     saveOptions = item.getSaveDialogOptions?() ? {}
     saveOptions.defaultPath ?= item.getPath()
-    newItemPath = atom.showSaveDialogSync(saveOptions)
+    showSaveDialogSync = if item.showSaveDialogSync else atom.showSaveDialogSync
```

Comment:
Sorry for the rubbish code, wanted to know if the change would be welcomed.
I'll add tests and documentation.

On Thu, 8 Oct 2015 04:18 Nathan Sobo <notifications@github.com> wrote:

> In src/pane.coffee
> <https://github.com/atom/atom/pull/9070#discussion_r41440873>:
>
> > @@ -514,7 +514,8 @@ class Pane extends Model
> >
> >      saveOptions = item.getSaveDialogOptions?() ? {}
> >      saveOptions.defaultPath ?= item.getPath()
> > -    newItemPath = atom.showSaveDialogSync(saveOptions)
> > +    showSaveDialogSync = if item.showSaveDialogSync else atom.showSaveDialogSync
>
> Yeah, this line of code doesn't look legit to me. It would have to use the
> ? operator as you said. Also it would need to be bound to the item...
>
> showSaveDialogSync = item.showSaveDialogSync?.bind(item) ? atom.showSaveDialogSync.bind(atom)
>
> Also, we'd need tests for this. I'd be open to adding a point of
> flexibility here but we need tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/atom/pull/9070/files#r41440873>.
>